src/pathwayvis/services/api.ts

```diff
@@ -62,25 +47,9 @@ export class APIService {
     });
   }
 
-  // @matyasfodor Impossible to understand this
-  // There's probably a library which does this
-  private _parseUrlParams(path: string): string[] {
-    return _.compact(_.map(path.split(/\W/), (param) => {
-      if (!(new RegExp('^\\d+$').test(param)) && param && (new RegExp('(^|[^\\\\]):' + param + '(\\W|$)').test(path))) {
-        return param;
-      }
-    }));
-  }
-
   private _handleParams(path: string, params: Object): RequestDetails {
-    const urlParamKeys = _.intersection(_.keys(params), this._parseUrlParams(path));
-
-    _.each(urlParamKeys, (key) => {
-      path = path.replace(':' + key, params[key]);
-      // @matyasfodor This mutates the input
-      delete params[key];
-    });
-
+    path = (new RegExp(":[a-zA-Z]+").test(path) ? path.replace(/:[a-zA-Z]+/,
```

Comment:
IIUC this one turns `api/:<variablename>/something` into `api/<variablename>/something`, right?

There's no need to test the existence of `":[a-zA-Z]+"`, you could just write

```
return {
  path: path.replace(/:[a-zA-Z]+/, params[String(path.match(":[a-zA-Z]+")).replace(":", "")],
  params,
}
```

But I'm not very happy with this solution. Why exactly do we need this? I'd prefer not to have any path manipulations except template literals. Did you find a case when it cannot be used?